src/track/track.h

```diff
@@ -78,14 +78,16 @@ class Track : public QObject {
     Q_PROPERTY(QString titleInfo READ getTitleInfo STORED false NOTIFY infoChanged)
     Q_PROPERTY(QDateTime sourceSynchronizedAt READ getSourceSynchronizedAt STORED false)
 
-    mixxx::FileAccess getFileAccess() const {
-        // Copying QFileInfo is thread-safe due to implicit sharing,
+    SecurityTokenPointer getFileAccessToken() const {
+        // Copying a QSharedPointer is thread-safe
         // i.e. no locking needed.
```

Comment:
why was this unsafe previously? `FileAccess` consists of a `FileInfo` and a `QSharedPointer`, both are thread-safe to copy.

Reply:
Yes, but the context switch may happen in between so that the you have the old FileInfo with a new access token.